### DISCUSSION TITLE
Request: Allow to store multiple Connection Settings

### DISCUSSION THREAD

**Discussion by 0lhi** [Syncplay/syncplay#603] [Ideas]
It would be very nice to store a variety of Connection settings, like in Mumble, that can each be accessed with a single click.

<img src="https://www.howtoforge.com/images/mumble_on_ubuntu_15_04/big/7.png" alt="Mumble Connection List">

In my opinion those connection profiles should store both, server and room, maybe listed as Room@Server.

**Comment by daniel-123** [comment]
System for doing something similar already exists, just not in the GUI. You can create .syncplay configuration files in any media folders you have and they will apply custom settings (like sever address) whenever you open Syncplay with a file from said folder or one further down in hierarchy. Basically like .htaccess works for Apache.

We never really got around to adding it to GUI for 2 reasons:

* It's additional work for rather niche feature
* System with per-folder configurations works very well. For my own use case - outright better than a dialog window.

That said, with some discussion about how exactly it should look like and function - this could be a feature worth thinking about if somebody is up for implementing it.

**Comment by Et0h** [comment]
This is a feature which was requested in #371 back in December 2020.  My response from the time is still relevant and formed the basis for what I put on https://syncplay.pl/guide/trouble/ in response to the question "I use multiple servers. How do I remember different settings for each server?".  That original response was as follows:

When it comes to profiles, different people might want different things to change. As such, there is no one simple profile system which would satisfy everyone. Adding additional UI elements might make it easier for some, but it could make it more confusing for others.

For Windows users there are three main ways to do a profile system that already work with the current version of Syncplay, and some work for other operating systems (some requiring appropriate tweaks):
(1) **Multiple shortcuts**. Create a shortcut to Syncplay per profile, and use the command line options to specify any value you want to change. See https://syncplay.pl/guide/client/#commandline for a list of command line switches, but the main ones are --host, --name, and --password
(2) **Multiple portable versions of Syncplay**. A portable version of Syncplay is just a version of Syncplay which has a syncplay.ini configuration file within the same folder as Syncplay.exe. You can copy and paste the Syncplay folder and have a different configuration file for each version of Syncplay.
(3) **Use per-folder configuration file**. If you separate your media into folders for each of your servers and have a syncplay.ini in each, then any settings you specify in that file will be applied if you right click on that file and select to open it with Syncplay. This is explained at https://syncplay.pl/guide/client/#advanced

I think for 99% of use cases, option (1) is perfectly fine, and for the rest (2) is perfectly adequate. If you give me a common use case where it isn't enough then that could potentially justify an additional command line option such as --config-path=[] to allow for people to specify multiple profiles. This would help keep the interface clean, allowing more advanced users to specify multiple profiles without it cluttering or confusing things for regular users.
